From 5456b9f896975750d3c7b8854d4c031b7b7409b2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sat, 31 Aug 2024 04:21:44 +0200
Subject: [PATCH] Set prgname to application ID

Using the application ID ensures that Wayland compositors could match the
window with the application and show the appropriate icon for them.
---
 src/eog-application.c | 1 +
 src/eog-window.c      | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/eog-application.c b/src/eog-application.c
index 4cd699cd..2afe1088 100644
--- a/src/eog-application.c
+++ b/src/eog-application.c
@@ -334,6 +334,7 @@ eog_application_startup (GApplication *application)
 	gtk_icon_theme_append_search_path (gtk_icon_theme_get_default (),
                                            EOG_DATA_DIR G_DIR_SEPARATOR_S "icons");
 
+	g_set_prgname (APPLICATION_ID);
 	gtk_window_set_default_icon_name (APPLICATION_ID);
 	g_set_application_name (_("Eye of GNOME"));
 
diff --git a/src/eog-window.c b/src/eog-window.c
index 154cd52e..c8697270 100644
--- a/src/eog-window.c
+++ b/src/eog-window.c
@@ -854,7 +854,7 @@ add_file_to_recent_files (GFile *file)
 	recent_data->description = NULL;
 	recent_data->mime_type = (gchar *) eog_util_get_content_type_with_fallback (file_info);
 	recent_data->app_name = EOG_RECENT_FILES_APP_NAME;
-	recent_data->app_exec = g_strjoin(" ", g_get_prgname (), "%u", NULL);
+	recent_data->app_exec = g_strdup ("eog %u");
 	recent_data->groups = groups;
 	recent_data->is_private = FALSE;
 
-- 
GitLab

